Source/WebCore/dom/make_names.pl

```diff
@@ -915,26 +1211,158 @@ sub printNamesCppFile
 
     if (keys %allTags) {
         my $tagsNamespace = $parameters{tagsNullNamespace} ? "nullAtom()" : "${lowercaseNamespacePrefix}NS";
-        printDefinitions($F, \%allTags, "tags", $tagsNamespace);
+        my $tagsNamespaceEnumValue = $parameters{tagsNullNamespace} ? "None" : $parameters{namespace};
+        printDefinitions($F, \%allTags, "tags", $tagsNamespace, $tagsNamespaceEnumValue);
     }
     if (keys %allAttrs) {
         my $attrsNamespace = $parameters{attrsNullNamespace} ? "nullAtom()" : "${lowercaseNamespacePrefix}NS";
-        printDefinitions($F, \%allAttrs, "attributes", $attrsNamespace);
+        my $attrsNamespaceEnumValue = $parameters{attrsNullNamespace} ? "None" : $parameters{namespace};
+        printDefinitions($F, \%allAttrs, "attributes", $attrsNamespace, $attrsNamespaceEnumValue);
     }
 
     print F "}\n\n} }\n\n";
     close F;
 }
 
+sub printDOMNamesCppFile
+{
+    my $cppPath = shift;
+    my $F;
+    open F, ">$cppPath";
+
+    printLicenseHeader($F);
+    print F "#include \"config.h\"\n";
+    print F "#include \"DOMNames.h\"\n";
+    print F "\n";
+    for my $namespace (sort keys %allCppNamespaces) {
+        print F "#include \"${namespace}Names.h\"\n";
+    }
+    print F "\n";
+    print F "namespace WebCore {\n";
+    print F "\n";
+    print F "static constexpr void* tagQualifiedNamePointers[] = {\n";
+    my %handledTags = ();
+    for my $tagKey (sort byTagNameOrder keys %allTags) {
+        my $cppNamespace = $allTags{$tagKey}{cppNamespace};
+        my $identifier = $allTags{$tagKey}{identifier};
+        print F "    &${cppNamespace}Names::${identifier}Tag,\n" unless $handledTags{$identifier};
+        $handledTags{$identifier} = 1;
+    }
+    print F "};\n";
+    print F "\n";
+    print F "static constexpr StringImpl::StaticStringImpl unadjustedTagNames[] = {\n";
+    for my $tagKey (sort byTagNameOrder keys %allTags) {
+        next if $allTags{$tagKey}{unadjustedTagEnumValue} eq "";
+        print F "    \"$allTags{$tagKey}{parsedTagName}\",\n";
+    }
+    print F "};\n";
+    print F "\n";
+    print F "void DOMNames::init() {\n";
+    print F "    static bool initialized = false;\n";
+    print F "    if (std::exchange(initialized, true))\n";
+    print F "        return;\n";
+    print F "\n";
+    print F "    s_tagNames.construct();\n";
+    print F "    AtomString* tagNamesEntry = s_tagNames->begin() + 2;\n";
```

Comment:
What does this magic 2 come from?

Reply:
It was due to skipping past element 0 (not used) and 1 (for `Unknown`). Per your previous comment, I'll make the `Unknown` values 0, which means this 2 will become a 1. And I'll add a comment in the generated code to indicate where that comes from.